components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -379,34 +396,35 @@ export const PortfolioAsset = (props: Props) => {
 
   const formattedFullAssetBalance = fullAssetBalances?.assetBalance
     ? new Amount(fullAssetBalances?.assetBalance ?? '')
-      .divideByDecimals(selectedAsset?.decimals ?? 18)
-      .formatAsAsset(6, selectedAsset?.symbol ?? '')
+      .divideByDecimals(selectedAssetFromParams?.decimals ?? 18)
+      .formatAsAsset(6, selectedAssetFromParams?.symbol ?? '')
     : ''
 
   const formattedAssetBalance = React.useMemo(() => {
     if (!fullAssetBalances?.assetBalance) return ''
 
     return new Amount(fullAssetBalances.assetBalance)
-      .divideByDecimals(selectedAsset?.decimals ?? 18)
+      .divideByDecimals(selectedAssetFromParams?.decimals ?? 18)
       .formatAsAsset(8)
-  }, [fullAssetBalances, selectedAsset])
+  }, [fullAssetBalances, selectedAssetFromParams])
 
   const isSelectedAssetDepositSupported = React.useMemo(() => {
-    if (!selectedAsset) {
+    if (!selectedAssetFromParams) {
       return false
     }
 
     return combinedTokensList.some(token =>
-      token.symbol.toLowerCase() === selectedAsset.symbol.toLowerCase())
-  }, [combinedTokensList, selectedAsset?.symbol])
+      token.symbol.toLowerCase() === selectedAssetFromParams.symbol.toLowerCase())
+  }, [combinedTokensList, selectedAssetFromParams?.symbol])
 
   // methods
   const onChangeTimeline = React.useCallback((timeline: BraveWallet.AssetPriceTimeframe) => {
+    setSelectedTimeline(timeline)
     dispatch(WalletPageActions.selectAsset({
-      asset: selectedAsset,
+      asset: selectedAssetFromParams,
       timeFrame: timeline
     }))
-  }, [selectedAsset])
+  }, [selectedAssetFromParams])
 
   const goBack = React.useCallback(() => {
     dispatch(WalletPageActions.selectAsset({ asset: undefined, timeFrame: selectedTimeline }))
```

Comment:
do we still need to dispatch all of these?

Reply:
@josheleonard I left there here to avoid breaking other parts of the wallet that are using the set values